### PR TITLE
Add EventBusClient and ResultDataClient interfaces

### PR DESCRIPTION
**Pull request by jreynard-code**
- Add ConditionalOnExpression on AzureDataExplorerClient
- Add ConditionalOnExpression on AzureEventHubsClient

### CODE REVIEW

File: src/main/kotlin/com/cosmotech/api/azure/adx/AzureDataExplorerClient.kt [hunk 1]
```diff
@@ -36,11 +37,11 @@ internal const val HEALTH_KUSTO_QUERY =
 
 @Service("csmADX")
 @Suppress("TooManyFunctions")
-@ConditionalOnProperty(name = ["csm.platform.vendor"], havingValue = "azure", matchIfMissing = true)
```
Comment (sjoubert) [inline src/main/kotlin/com/cosmotech/api/azure/adx/AzureDataExplorerClient.kt:39]: Do we have another vendor or do we assume `vendor=azure` when `use-internal-result-services=false`? Should we keep the conditional property also?

(Same on the second service)

Comment (neomatamune) [reply src/main/kotlin/com/cosmotech/api/azure/adx/AzureDataExplorerClient.kt:39]: I'm pretty sure we won't really be going to go back to ADX + Eventhub on azure once we started using the internal result services, so we can assume for the ResultDataClient and EventBusClient that the only external service should be on Azure and nowhere else.

Comment (jreynard-code) [reply src/main/kotlin/com/cosmotech/api/azure/adx/AzureDataExplorerClient.kt:39]: Right now we can assume that ADXClient or EventHubClient will ve used only if `useinternalresultservices` is not true. If we have a case in future with ADX/EventHub Client and result data services, we will change the conditional at this moment. Same thing if we have a case with AWS/GCP/....

Comment (sjoubert) [reply src/main/kotlin/com/cosmotech/api/azure/adx/AzureDataExplorerClient.kt:39]: :+1:

**Pull request merged**